src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const instuction = Object.assign(actions);
+  const loopResult = { ...state };
+
+  for (let i = 0; i < instuction.length; i++) {
+    switch (instuction[i].type) {
+      case 'addProperties':
+        for (const key in instuction[i].extraData) {
```

Comment:
Consider using Object.assign here for better performance

Reply:
Sorry, I didn't understand what you meant.